datalake-api/src/main/java/com/sequenceiq/sdx/api/endpoint/SdxEndpoint.java

```diff
@@ -204,6 +206,21 @@ SdxBackupStatusResponse backupDatalakeStatusByName(@PathParam("name") String nam
             @QueryParam("backupId") String backupId,
             @QueryParam("backupName") String backupName);
 
+    @POST
+    @Path("{name}/restoreDatalake")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "restore the datalake ", produces = MediaType.APPLICATION_JSON, nickname = "restoreDatalake")
+    SdxRestoreResponse restoreDatalakeByName(@PathParam("name") String name,
+            @QueryParam("backupId") String backupId, @QueryParam("backupLocation") String backupLocation,
+            @QueryParam("backupLocationOverride") String backupLocationOverride);
```

Comment:
usage: cdp datalake restore-datalake --datalake-name DATALAKE_NAME
                                     [--backup-id BACKUP_ID]
                                     [--backup-name BACKUP_NAME]
                                     [--include-database]
                                     [--no-include-database] [--force]
                                     [--no-force]
                                     [--backup-location-override BACKUP_LOCATION_OVERRIDE]
                                     [--cli-input-json CLI_INPUT_JSON]
                                     [--generate-cli-skeleton]
                                     [help]




The backup-location-override is optional incase user put backup in a separate location. Backup location is not null because it will be used to fetch database backup from the s3. It is in the cdpcli. Let's keep it this way. and if needed, Let's rename cdpcli first. This jira is for add the flow to cloudbreak. any protocol change should be done with cdpcli first and with a separate jira.

Reply:
removed backuplocatoin field. Thanks!